include/deal.II/base/function.templates.h

```diff
@@ -297,240 +297,200 @@ Function<dim, Number>::memory_consumption () const
 
 //---------------------------------------------------------------------------
 
-
-
-template <int dim, typename Number>
-ZeroFunction<dim, Number>::ZeroFunction (const unsigned int n_components)
-  :
-  Function<dim, Number> (n_components)
-{}
-
-
-template <int dim, typename Number>
-ZeroFunction<dim, Number>::~ZeroFunction ()
-{}
-
-
-template <int dim, typename Number>
-Number ZeroFunction<dim, Number>::value (const Point<dim> &,
-                                         const unsigned int) const
-{
-  return 0.;
-}
-
-
-template <int dim, typename Number>
-void ZeroFunction<dim, Number>::vector_value (const Point<dim> &,
-                                              Vector<Number>   &return_value) const
+namespace Functions
 {
-  Assert (return_value.size() == this->n_components,
-          ExcDimensionMismatch (return_value.size(), this->n_components));
 
-  std::fill (return_value.begin(), return_value.end(), 0.0);
-}
+  template <int dim, typename Number>
+  ZeroFunction<dim, Number>::ZeroFunction (const unsigned int n_components)
+    :
+    ConstantFunction<dim, Number> (Number(), n_components)
+  {}
 
 
-template <int dim, typename Number>
-void ZeroFunction<dim, Number>::value_list (
-  const std::vector<Point<dim> > &points,
-  std::vector<Number>            &values,
-  const unsigned int              /*component*/) const
-{
-  (void)points;
-  Assert (values.size() == points.size(),
-          ExcDimensionMismatch(values.size(), points.size()));
+  template <int dim, typename Number>
+  ZeroFunction<dim, Number>::~ZeroFunction ()
+  {}
```

Comment:
since the destructor is already `virtual` in the base class, there is need to declare or implement one here -- just remove it.